warden/x/async/keeper/futures.go

```diff
@@ -0,0 +1,77 @@
+package keeper
+
+import (
+	"context"
+	"fmt"
+
+	"cosmossdk.io/collections"
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/warden-protocol/wardenprotocol/warden/repo"
+	types "github.com/warden-protocol/wardenprotocol/warden/x/async/types/v1beta1"
+)
+
+type FutureKeeper struct {
+	futures         repo.SeqCollection[types.Future]
+	futureByCreator collections.KeySet[collections.Pair[sdk.AccAddress, uint64]]
+	results         collections.Map[uint64, types.FutureResult]
+}
+
+func NewFutureKeeper(sb *collections.SchemaBuilder, cdc codec.Codec) *FutureKeeper {
+	futuresSeq := collections.NewSequence(sb, FuturesPrefix, "futures_sequence")
+	futuresColl := collections.NewMap(sb, FutureSeqPrefix, "futures", collections.Uint64Key, codec.CollValue[types.Future](cdc))
```

Comment:
I think this have to be changed  to
```suggestion
	futuresColl := collections.NewMap(sb, FuturePrefix, "futures", collections.Uint64Key, codec.CollValue[types.Future](cdc))
```